gp-templates/helper-functions.php

```diff
@@ -156,8 +169,26 @@ function map_glossary_entries_to_translation_originals( $translation, $glossary,
 		}
 	}
 
-	// Split the singular string on word boundaries.
-	$singular_split    = preg_split( '/\b/', $translation->singular );
+	// Sort glossary entries by word count, to search first for multiple word terms, words with hyphens or both.
+	usort(
+		$glossary_entries_terms_array,
+		function( $a, $b ) {
+			$a_words = preg_split( '/[\s-]/', $a );
+			$b_words = preg_split( '/[\s-]/', $b );
+			return count( $b_words ) <=> count( $a_words );
+		}
+	);
+
+	// Set terms search string.
+	$terms_search = array();
+	foreach ( $glossary_entries_terms_array as $term ) {
+		// Add word boundaries to search term.
+		$terms_search[] = '(\b' . $term . '\b)';
```

Comment:
Is there a `preg_quote()` missing?

Reply:
There was already a preg_quote() on https://github.com/GlotPress/GlotPress-WP/blob/develop/gp-templates/helper-functions.php#L89 that processes all the terms in the array. Would it be better to use it immediatly before the search?

Reply:
I can edit it and only preg_quote immediatly before the search, but have to remove all the preg_quote()s on gp_sort_glossary_entries_terms().
It keeps the code simpler, should I go for it?

Reply:
I've done it in https://github.com/GlotPress/GlotPress-WP/pull/1359/commits/f5b86cdf21279213fa2247986484261dfb1e3bda
Looks cleaner, let me know if it's ok for you.

Reply:
Yes, thank you!